### ISSUE TITLE
comparison_chain warning even though being exhaustive is not desired

### ISSUE THREAD

**Issue by ArneCode** [rust-lang/rust-clippy#9147]
### Summary

I think clippy shouldn't give me a warning in this case.

### Lint Name

comparison_chain

### Reproducer

I tried this code:

```rust
fn main() {
    clippy(0);
}
fn clippy(level: i32){
    if level==0{
        println!("is zero");
    }else if level < 0{
        println!("less than zero");
    }
}
```

I saw this happen:

```
warning: `if` chain can be rewritten with `match`

  --> src/main.rs:6:5
   |
6  | /     if level==0{
7  | |         println!("is zero");
8  | |     }else if level < 0{
9  | |         println!("less than zero");
10 | |     }
   | |_____^
   |
   = note: `#[warn(clippy::comparison_chain)]` on by default
   = help: consider rewriting the `if` chain to use `cmp` and `match`
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#comparison_chain
```

I expected to see this happen:
I wouldn't want to be exhaustive in this case. Of course I could rewrite the code to be exhaustive but I only want a certain actions executed in these two cases.
I am still kind of new to rust and clippy so I am sorry if I am understanding it wrong

### Version

```text
rustc 1.62.0
binary: rustc
commit-hash: unknown
commit-date: unknown
host: aarch64-linux-android
release: 1.62.0
LLVM version: 14.0.6
```

**Comment by Jarcho** [comment]
The equivalent match would be
```rust
match level.cmp(0) {
    Ordering::Equal => println!("is zero"),
    Ordering::Less => println!("less than zero"),
    Ordering::Greater => (),
}
```

**Comment by ArneCode** [comment]
Ok, thanks

**Comment by dhardy** [comment]
The justification for this lint is:

> `if` is not guaranteed to be exhaustive and conditionals can get repetitive

When there are only two cases it seems clear that the `if` is not intended to be exhaustive while the second point "conditionals can get repetitive" seems moot.

In my opinion, this lint probably makes sense for `if a < b { .. } else if a == b { .. } else { .. }` but not `if a < b { .. } else if a == b { .. }` (where there is no third case).
